producer/writer/message_writer.go

```diff
@@ -334,9 +323,9 @@ func (w *messageWriterImpl) retryBatchWithLock(
 			w.mPool.Put(m)
 			continue
 		}
-		if w.skipRetry {
+		if w.isClosed {
```

Comment:
Why are we changing from "optionally dropping everything" to "always drop everything"?

Reply:
sounds good